apis/duck/cached_test.go

```diff
@@ -105,28 +128,49 @@ func TestDifferentGVRs(t *testing.T) {
 			Version:  fmt.Sprintf("v%d", i),
 			Resource: "caches",
 		}
-		grp.Go(func() error {
+
+		errGrp.Go(func() error {
 			_, _, err := cif.Get(gvr)
-			atomic.AddInt32(&returned, 1)
+			atomic.AddInt32(&retGetCount, 1)
 			return err
 		})
 	}
 
 	// Give the goroutines time to make progress.
 	time.Sleep(100 * time.Millisecond)
 
-	// Check that none have returned and we have 10 Gets in progress.
-	if got, want := atomic.LoadInt32(&returned), int32(0); got != want {
-		t.Errorf("Got %v returned, wanted %v", got, want)
+	// Check that no call to cif.Get have returned and bif.Get was called
+	// once per iteration.
+	if got, want := atomic.LoadInt32(&retGetCount), int32(0); got != want {
+		t.Errorf("Got %d returned call(s) to cif.Get, wanted %d", got, want)
 	}
-	if got, want := atomic.LoadInt32(&bif.getting), int32(10); got != want {
-		t.Errorf("Got %v calls to bif.Get, wanted %v", got, want)
+	if got, want := atomic.LoadInt32(&bif.nCalls), int32(iter); got != want {
+		t.Errorf("Got %d call(s) to bif.Get, wanted %d", got, want)
 	}
 
 	// Allow the Get calls to proceed.
 	close(bif.block)
 
-	if err := grp.Wait(); err != nil {
-		t.Errorf("Wait() = %v", err)
+	if err := errGrp.Wait(); err != nil {
+		t.Fatalf("Error while calling cif.Get: %v", err)
+	}
+
+	// Check that all calls to cif.Get have returned and the number of
+	// calls to bif.Get didn't increase.
+	if got, want := atomic.LoadInt32(&retGetCount), int32(iter); got != want {
+		t.Errorf("Got %d returned call(s) to cif.Get, wanted %d", got, want)
+	}
+	if got, want := atomic.LoadInt32(&bif.nCalls), int32(iter); got != want {
+		t.Errorf("Got %d call(s) to bif.Get, wanted %d", got, want)
+	}
```

Comment:
```suggestion
	if got, want := atomic.LoadInt32(&retGetCount), iter; got != want {
		t.Errorf("Got %d returned call(s) to cif.Get, wanted %d", got, want)
	}
	if got, want := atomic.LoadInt32(&bif.nCalls), iter; got != want {
		t.Errorf("Got %d call(s) to bif.Get, wanted %d", got, want)
	}
```

Reply:
You can't do that:
```
invalid operation: got != want (mismatched types int32 and int)
```

Reply:
Ah, we need to do `int32(iter)` I guess? But 🤷 probably doesn't matter.

Reply:
`const iter int32 = 10` would work yes, I can add it if you feel strongly about it,

Reply:
(but then you also need an int32 here: `for i := 0; i < iter; i++ {`)

Reply:
yeah, that's fine doesn't matter much. thanks!